Tasks/AzureRmWebAppDeployment/xdtutility.ts

```diff
@@ -0,0 +1,48 @@
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
+
+function applyXmlTransformation(sourceFile, transformFile) {
+	var cttPath = path.join(__dirname, 'ctt/ctt.exe');
+    var cttArgs = ' s:"' + sourceFile + '" t:"' + transformFile + '" d:"' + sourceFile + '" pw';
+    var cttExecutionResult = tl.execSync(cttPath, cttArgs, <trm.IExecOptions>{failOnStdErr: true, silent: true});
+    if(cttExecutionResult.stderr)
+        throw "XML Transformation Error while transforming " + sourceFile + " using " + transformFile;
+}
+
+export function xmlTransform(transformXmlPattern : string, sourceXmlPattern : string) {
+	var transformXmlFiles = expandWildcardPattern (transformXmlPattern);
+	var sourceXmlFiles = expandWildcardPattern (sourceXmlPattern);
+
+	var transformXmlPatternBaseName = getPatternBaseName (transformXmlPattern);
+	var sourceXmlPatternBaseName = getPatternBaseName (sourceXmlPattern);
+
+	Object.keys(sourceXmlFiles).forEach(function(xmlFile) {
+		var transformXmlFile = path.join(path.dirname(xmlFile), path.basename(xmlFile, sourceXmlPatternBaseName) + transformXmlPatternBaseName);
+		if (transformXmlFiles[transformXmlFile]) {
+			tl._writeLine('Applying XML Transformation : ' + transformXmlFile + '->' + xmlFile);
+			applyXmlTransformation(xmlFile, transformXmlFile);
+		}
+	});
+}
+
+function getPatternBaseName (filePattern : string) : string {
+	// returns the substring from the last found wildcard character to the end of the filePattern string
+	var fileBaseName = path.basename(filePattern);
+	var wildcardChars = '?_*%#]';
+	for (var i = fileBaseName.length - 1; i >= 0; i--) {
```

Comment:
Is this safe to assume that wildcards are not interspread? Ex - pathdir/projecta*.config and pathdir/projecta*.release.config? This utility cannot assume the specific input that you are passing now. This will have to support additional files soon.